src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -151,6 +151,11 @@ internal partial class XamlFileGenerator
 		/// </summary>
 		private readonly bool _shouldAnnotateGeneratedXaml;
 
+		/// <summary>
+		/// Used to detect duplicate x:Name and report error for those.
+		/// </summary>
+		private HashSet<string> _names = new HashSet<string>();
```

Comment:
I think x:Name validations must be scoped to the proper "namescope". For instance, it's  valid to have two `DataTemplate` which contain the same `x:Name` member.